### PR TITLE
feat(es/parser): Add `disallowAssertKeywords` option

### PR DESCRIPTION
**Pull request by petamoriken**
**Description:**

V8 will remove support for the `assert` keywords, so Node.js and Chrome will follow suit.

> Hey, quick update on this. Node.js is planning to remove support for `assert` in v22 (which will be released in April) and Chrome in v126 (which will be released in May).

https://github.com/denoland/deno/issues/17944#issuecomment-2027005952

Disallow the `assert` keywords in Deno by adding the `disallowAssertKeywords` option in this PR.

**BREAKING CHANGE:**

N/A

**Related issue (if exists):**

https://github.com/swc-project/swc/issues/8893#issuecomment-2080369150

### CODE REVIEW

**Comment by petamoriken** [comment]
@kdy1 Where should I write the test codes?

**Comment by kdy1** [comment]
You can add a test like https://github.com/swc-project/swc/blob/c1115a2ebbca4fdfb2d4dc589c2d82b878cf9f88/crates/swc_ecma_parser/src/parser/tests.rs#L321-L345

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_parser --breaking

File: crates/swc_ecma_parser/src/lib.rs [hunk 1]
```diff
@@ -187,6 +187,20 @@ impl Syntax {
...
 
+    pub fn disallow_assert_keywords(self) -> bool {
+        match self {
+            Syntax::Es(EsConfig {
+                disallow_assert_keywords,
+                ..
+            }) => disallow_assert_keywords,
+            #[cfg(feature = "typescript")]
+            Syntax::Typescript(TsConfig {
+                disallow_assert_keywords,
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/lib.rs:198]: Please remove this

File: crates/swc_ecma_parser/src/lib.rs [hunk 2]
```diff
@@ -315,6 +329,9 @@ pub struct TsConfig {
     #[serde(default)]
     pub decorators: bool,
 
+    #[serde(default)]
+    pub disallow_assert_keywords: bool,
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/lib.rs:333]: and this

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_parser/src/lib.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_parser/src/lib.rs:198]: We should unconditionally follow the behavior of tsc

Comment (petamoriken) [reply crates/swc_ecma_parser/src/lib.rs:198]: Oh, I see. The issue I want to solve with this PR is that it prevents the `assert` keywords from being used in TypeScript, so I'll try to confirm if tsc can handle it.

**Comment by petamoriken** [comment]
Since tsc has not decided on a direction, this pull request is closed for the time being.

**Pull request closed without merging**